src/App.tsx

```diff
@@ -1,61 +1,147 @@
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+
+          if (!newTitle || !newTitle.match(/^[\sа-яіїєa-z0-9]+$/i)
+            || onlySpaces(newTitle)) {
+            setTitleError(true);
+          }
+
+          if (!selectedUserId) {
+            setUserIdError(true);
+          }
+
+          if (!newTitle || !newTitle.match(/^[\sа-яіїєa-z0-9]+$/i)
+            || onlySpaces(newTitle) || !selectedUserId) {
+            return;
+          }
+
+          setTodos((prevTodos) => {
+            return [
+              ...prevTodos,
+              {
+                title: newTitle,
+                userId: selectedUserId,
+                completed: false,
+                user: getUser(selectedUserId),
+                id: prevTodos
+                  .map(({ id }) => id)
+                  .reduce((idA, idB) => Math.max(idA, idB), 0) + 1,
+              },
+            ];
+          });
+          setTitle('');
+          setselectedUserId(0);
+        }}
+      >
+
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title: '}
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTitle}
+              onChange={(event) => {
+                setTitle(event.target.value);
+                if (event.target.value.match(/^[\sа-яіїєa-z0-9]+$/i)) {
+                  setTitleError(false);
+                } else {
+                  setTitleError(true);
+                }
+              }}
+            />
+            {titleError
+              && (
+                <>
+                  <span className="error">Please enter a title</span>
+                  <span className="error_extension">
+                    {' '}
+                    Title should include UA or EN letters,
+                    numbers and spaces.
+                  </span>
+                </>
+              )}
+
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            {'User: '}
+            <select
+              value={selectedUserId}
+              data-cy="userSelect"
+              onChange={(event) => {
+                setUserIdError(false);
+                setselectedUserId(Number(event.target.value));
+              }}
+            >
+              <option value="0" disabled>
+                Choose a user
+              </option>
+              {usersFromServer.map((user) => (
+                <option
+                  value={user.id}
```

Comment:
you forgot to add "key"

Reply:
thanks! I thought that value as user.id will work as key. But I will add key)